From tj@kernel.org  Mon Nov 17 15:00:43 2008
From: Tejun Heo <tj@kernel.org>
Date: Thu, 13 Nov 2008 13:20:00 +0900
Subject: uevent: don't pass envp_ext[] as format string in kobject_uevent_env()
To: Greg KH <greg@kroah.com>, Kay Sievers <kay.sievers@vrfy.org>
Message-ID: <491BAAF0.5010203@kernel.org>

From: Tejun Heo <tj@kernel.org>

kobject_uevent_env() uses envp_ext[] as verbatim format string which
can cause problems ranging from unexpectedly mangled string to oops if
a string in envp_ext[] contains substring which can be interpreted as
format.  Fix it.

Signed-off-by: Tejun Heo <tj@kernel.org>
Cc: Kay Sievers <kay.sievers@vrfy.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 lib/kobject_uevent.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/lib/kobject_uevent.c
+++ b/lib/kobject_uevent.c
@@ -165,7 +165,7 @@ int kobject_uevent_env(struct kobject *k
 	/* keys passed in from the caller */
 	if (envp_ext) {
 		for (i = 0; envp_ext[i]; i++) {
-			retval = add_uevent_var(env, envp_ext[i]);
+			retval = add_uevent_var(env, "%s", envp_ext[i]);
 			if (retval)
 				goto exit;
 		}
